backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/project/workorder/WorkOrderProjectResourceRestService.java

```diff
@@ -504,11 +507,17 @@ private static JsonWorkOrderResourceResponse toJsonWorkOrderResourceResponse(
 	@Value(staticConstructor = "of")
 	private static class WOProjectResourceIdentifier
 	{
-
 		@NonNull
 		WOProjectStepId stepId;
 
 		@NonNull
 		IdentifierString resourceIdentifier;
 	}
+
+	@NonNull
+	private static Predicate<WOProjectResourceIdentifier> distinctByResourceIdentifier(@NonNull final Function<WOProjectResourceIdentifier, IdentifierString> keyExtractor)
```

Comment:
it took me a few minutes to wrap my had around this...i guess it's legit though. wdyt, @teosarca

Reply:
LE/sidenote: it took me another few minutes to believe that there is no `distinctBy(keyextrator)` method in  java.util..but it seems there isn't